### PR TITLE
Discovery: Error if properties are missing

### PR DESCRIPTION
**Pull request by ckamm**
This is potentially too aggressive? What do you think @ogoffart?

See #6317

### CODE REVIEW

File: src/libsync/discoveryphase.cpp [hunk 1]
```diff
@@ -374,12 +374,26 @@ void DiscoverySingleDirectoryJob::directoryListingIteratedSlot(QString file, con
             file = file.remove(0, 1);
         }
 
-
-        std::unique_ptr<csync_file_stat_t> file_stat(propertyMapToFileStat(map));
+        std::unique_ptr<csync_file_stat_t> file_stat(new csync_file_stat_t);
         file_stat->path = file.toUtf8();
-        if (file_stat->etag.isEmpty()) {
-            qCCritical(lcDiscovery) << "etag of" << file_stat->path << "is" << file_stat->etag << "This must not happen.";
+        file_stat->size = -1;
```
Comment (ogoffart) [inline src/libsync/discoveryphase.cpp]: Should we give some default value in csync_file_stat_s

File: src/libsync/discoveryphase.cpp [hunk 2]
```diff
@@ -374,12 +374,26 @@ void DiscoverySingleDirectoryJob::directoryListingIteratedSlot(QString file, con
...
+        if (file_stat->type == ItemTypeDirectory)
+            file_stat->size = 0;
+        if (file_stat->type == ItemTypeSkip
+            || file_stat->size == -1
+            || file_stat->modtime == -1
+            || file_stat->remotePerm.isNull()
+            || file_stat->etag.isEmpty()
+            || file_stat->file_id.isEmpty()) {
+            _error = tr("The server file discovery reply is missing data.");
+            qCCritical(lcDiscovery)
```
Comment (ogoffart) [inline src/libsync/discoveryphase.cpp]: i think qCWarning should be enough compared to the other reason we use qcWarning

**Comment by ogoffart** [comment]
The comment for `OwncloudPropagator::isInSharedDirectory`
```
// ownCloud server  < 7.0 did not had permissions
```

If we no longer support these server, it is a bunch of code we can remove.

**Comment by ckamm** [comment]
Reading the fileid from PROPFIND was introduced in 4440acdb1de37ade4a8ef24cf97f8ab0915c1d33 (Oct 2013).

**Comment by ckamm** [comment]
@ogoffart The current state as far as I am aware is that the client considers server versions before 7.0 as unsupported. The client shows a "Unsupported server version, proceed at your own risk" message with these. Requiring permissions support would break them. Let's consider whether to drop pre-7 support for 2.5?

**Comment by ogoffart** [comment]
i agree, we probably need not to support these server. https://github.com/owncloud/client/pull/6380

**Comment by PVince81** [comment]
was there a policy that says that clients must support EOL'd versions of OC, even very old ones ?

**Comment by michaelstingl** [comment]
> was there a policy that says that clients must support EOL'd versions of OC, even very old ones ?

I'm not aware. Why should this be the case?

**Comment by PVince81** [comment]
just to answer this: https://github.com/owncloud/client/pull/6379#issuecomment-369512277

if we don't need to support EOL'd versions of OC then these versions could be blocked completely by the client ?

**Comment by ogoffart** [comment]
@PVince81  done in #6380

**Pull request merged**